### PR TITLE
Add Context Menu to Disassembly View

### PR DESCRIPTION
**Pull request by yuehuang010**
Added context menu to Disassembly View to have:
 - Set Instruction Breakpoint (F9)
 - Toggle Source Code
 - Go to Disassembly Address
 - Copy Disassembly as JSON

Added QuickPicker for GoToDisassembly as '##'.  Open for discuss on a better symbol.

image

### CODE REVIEW

**Comment by yuehuang010** [comment]
FYI. @isidorn @roblourens @weinand

**Comment by yuehuang010** [comment]
Ping. @roblourens

File: src/vs/workbench/contrib/debug/browser/debugEditorActions.ts [hunk 1]
```diff
@@ -41,17 +50,50 @@ class ToggleBreakpointAction extends EditorAction2 {
...
+	async run(accessor: ServicesAccessor, ...args: any[]) {
+		// Find the editor with text focus or active
+		const codeEditorService = accessor.get(ICodeEditorService);
+		const editor = codeEditorService.getFocusedCodeEditor() || codeEditorService.getActiveCodeEditor();
+		if (!editor) {
+			// maybe it is disassemblyView.
+			const editorService = accessor.get(IEditorService);
+			if (editorService.activeEditor?.typeId === DisassemblyViewInput.ID) {
+				const view = editorService.activeEditorPane as DisassemblyView;
+				const selection = view?.getSelection();
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debugEditorActions.ts:78]: These work on the selected row, not the one I right-clicked. You probably want the focused row, not selected.

File: src/vs/workbench/contrib/debug/browser/disassemblyView.ts [hunk 2]
```diff
@@ -177,19 +187,19 @@ export class DisassemblyView extends EditorPane {
...
+				this._instantiationService.createInstance(InstructionRenderer, this),
 			],
 			{
 				identityProvider: { getId: (e: IDisassembledInstructionEntry) => e.instruction.address },
 				horizontalScrolling: false,
 				overrideStyles: {
 					listBackground: editorBackground
 				},
-				multipleSelectionSupport: false,
+				multipleSelectionSupport: true,
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/disassemblyView.ts:198]: Why multi-select support?

File: src/vs/workbench/contrib/debug/browser/disassemblyView.ts [hunk 3]
```diff
@@ -177,19 +187,19 @@ export class DisassemblyView extends EditorPane {
 			],
 			[
 				this._instantiationService.createInstance(BreakpointRenderer, this),
-				instructionRenderer,
+				this._instantiationService.createInstance(InstructionRenderer, this),
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/disassemblyView.ts:190]: Do these renderers need to be disposed?

File: src/vs/workbench/contrib/debug/browser/debugEditorActions.ts [hunk 4]
```diff
@@ -200,6 +247,105 @@ class ToggleDisassemblyViewSourceCodeAction extends Action2 {
...
+}
+
+class DisassemblyGotoAddressQuickAccessProvider implements IQuickAccessProvider {
+	static PREFIX: string = '##';
+	defaultFilterValue?: string | DefaultQuickAccessFilterValue | undefined = DefaultQuickAccessFilterValue.PRESERVE;
+
+	constructor(@IEditorService private readonly editorService: IEditorService) {
+	}
+
+	provide(picker: IQuickPick<IQuickPickItem>, token: CancellationToken): IDisposable {
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debugEditorActions.ts:293]: This is not working for me. Nothing happens when I press enter

**Review by roblourens — CHANGES_REQUESTED** [review]

**Comment by roblourens** [comment]
@yuehuang010 are you planning on working on this?

**Comment by yuehuang010** [comment]
> @yuehuang010 are you planning on working on this?

No.  @xisui-MSFT are you interested in picking it up?

**Comment by roblourens** [comment]
Ping @xisui-MSFT is any work going to happen here? If not I will close it

**Comment by xisui-MSFT** [comment]
@roblourens It's on our backlog, but I don't have time to work on it right now. Its base is already pretty outdated anyway, so I think you can close it, and if we are able to continue working on it in the future, we can open another PR.

**Pull request closed without merging**